### ISSUE TITLE
Bring nested recursion example into noir-lang/noir

### ISSUE THREAD

**Issue by Savio-Sou** [noir-lang/noir#4410]
```[tasklist]
### Prerequisites
- [ ] https://github.com/AztecProtocol/dev-rel/pull/207
```

**Comment by jzaki** [comment]
Details from Tom:
- [ ] We'll need to have them pull the dependencies from the workspace rather than being fixed to a particular npm version.
  - If we're referencing the package.json in the docs then we're going to need to swap out the dependencies for the npm versions when pushing a new release.
- [ ] We're going to need to run these as part of CI so we can enforce that the code still works and we can fix as needed.

**Comment by Savio-Sou** [comment]
Following https://github.com/noir-lang/noir/pull/5072, proving and verification workflow would be scoped out of NoirJS.

The PR includes updates to the doc page. Commenting here for @noir-lang/developerrelations and readers' awareness in case it leads to changes to existing / planned works with respect to this Issue.

**Comment by critesjosh** [comment]
if not done by the end of the week, moving to blocked

**Comment by jzaki** [comment]
Yeah, issue relayed again to Noir team. Now blocked until workaround/fix found.

**Comment by Savio-Sou** [comment]
@jzaki could you link me to the relevant noir-lang/noir or AztecProtocol/aztec-packages Issue?

**Comment by jzaki** [comment]
Okay, with a little tip form Tom, I did get it to pass, but by doing something that seems wrong and different to between what nargo returns and bb expects between binary vs js. I'll check with Tom this morning and create an issue for the lack of intuition on the .js side to document, or an issue for the .js side to be aligned with the binarys' way.

**Comment by jzaki** [comment]
Added two benchmark circuits from Savio, 2^17 succeeds, 2^18 fails. The recursion test is skipped.
PR [here](https://github.com/noir-lang/noir/pull/4969)

Once 2^18 works again, the skipped recursion tests should readily be made to pass again, so can unskip it and remove the benchmark tests.

**Comment by Savio-Sou** [comment]
Thanks @jzaki. If you have time it might be good to file an Issue for it in https://github.com/AztecProtocol/aztec-packages.
